ui/src/app/shared/services/projects-service.ts

```diff
@@ -95,22 +95,30 @@ export class ProjectsService {
         return requests.get(`/projects/${name}`).then(res => res.body as models.Project);
     }
 
-    public delete(name: string): Promise<boolean> {
-        return requests.delete(`/projects/${name}`).then(() => true);
+    public async delete(name: string): Promise<boolean> {
+        const csrfToken = await requests.getCsrfToken();
```

Comment:
Hm, the only other way to supply the token to the user's client would be via hidden form fields. The question is, how to get the token. If we would template out the JS from the API server, this would be easy. But we don't, the UI code is just static assets, served via HTTP handler from our server.

Maybe switching to HEAD as discussed above would remedy the situation a little. However, this affects requests that manipulate data (POST, PUT, DELETE currently), which is probably the minority of calls to the API compared to data retrieval GET queries (which will not be doubled).